core/lib/rom/schema.rb

```diff
@@ -62,6 +62,8 @@ class Schema
 
     DEFAULT_INFERRER = Inferrer.new(enabled: false).freeze
 
+    VALID_IVAR_REGEX = /^[a-z_][a-zA-Z_0-9]*$/
```

Comment:
Why must the instance variable have a lowercase letter first?  This seems to work fine:

```rb
2.6.3 :001 > Object.new.instance_variable_set :@A, :foo
 => :foo
2.6.3 :002 > "A".match( /^[a-z_][a-zA-Z_0-9]*$/)
 => nil
```

Would it make sense to catch the `NameError` raised by Ruby instead?
```rb
2.6.3 :005 > Object.new.instance_variable_set :"@A bad choice", :foo
Traceback (most recent call last):
        6: from /Users/gregorykohler/.rvm/rubies/ruby-2.6.3/bin/irb:23:in `<main>'
        5: from /Users/gregorykohler/.rvm/rubies/ruby-2.6.3/bin/irb:23:in `load'
        4: from /Users/gregorykohler/.rvm/rubies/ruby-2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        3: from (irb):5
        2: from (irb):5:in `rescue in irb_binding'
        1: from (irb):5:in `instance_variable_set'
NameError (`@A bad choice' is not allowed as an instance variable name)
```

(I haven't found where in the C code the validity is defined)